packages/components/tests/integration/components/hds/modal/index-test.js

```diff
@@ -0,0 +1,221 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { click, render, resetOnerror, setupOnerror } from '@ember/test-helpers';
+import { hbs } from 'ember-cli-htmlbars';
+
+module('Integration | Component | hds/modal/index', function (hooks) {
```

Comment:
general comment on these tests: now that the `<M.Header>` is not required anymore, you may remove it from these tests `render()` too (where it's not needed, of course).

Reply:
I remember what happened here. While the header is not required, `ember-focus-trap` needs an interactive element to be present, so I used the header (which always renders the close button) consistently across the examples to avoid an error from `ember-focus-trap`. We don't recommend using the modal without a dismiss action at the least (be that in the header or in the actions), so I don't think we'll ever see this issue in production but do let me know if you'd like me to handle this differently.

Reply:
Makes sense. Maybe in this case add a comment somewhere, explaining this special case?

Reply:
yep, will do now

Reply:
[Tried to cover this](https://github.com/hashicorp/design-system/pull/631/commits/15945680222593a844b4877bca703ca96d24cb31) by explaining how we use this Ember modifier, what are the requirements, and how we achieve them